aspnetcore/host-and-deploy/aspnet-core-module.md

```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
+
+If `stdoutLogEnabled` is false, errors that occur on app startup are captured and emitted to the event log up to 30 KB. After startup, all additional logs are discarded.
+
+The following sample `aspNetCore` element configures stdout logging for an app hosted in Azure App Service. A local path or network share path is acceptable for local logging. Confirm that the AppPool user identity has permission to write to the path provided.
+
+```xml
+<aspNetCore processPath="dotnet"
+    arguments=".\MyApp.dll"
+    stdoutLogEnabled="true"
+    stdoutLogFile="\\?\%home%\LogFiles\stdout"
+    hostingModel="InProcess">
+</aspNetCore>
+```
+
+## Enhanced diagnostic logs
+
+The ASP.NET Core Module is configurable to provide enhanced diagnostics logs. Add the `<handlerSettings>` element to the `<aspNetCore>` element in *web.config*. Setting the `debugLevel` to `TRACE` exposes a higher fidelity of diagnostic information:
+
+```xml
+<aspNetCore processPath="dotnet"
+    arguments=".\MyApp.dll"
+    stdoutLogEnabled="false"
+    stdoutLogFile="\\?\%home%\LogFiles\stdout"
+    hostingModel="InProcess">
+  <handlerSettings>
+    <handlerSetting name="debugFile" value=".\logs\aspnetcore-debug.log" />
+    <handlerSetting name="debugLevel" value="FILE,TRACE" />
+  </handlerSettings>
+</aspNetCore>
+```
+
+Folders in the path provided to the `<handlerSetting>` value (*logs* in the preceding example) aren't created by the module automatically and should pre-exist in the deployment. The app pool must have write access to the location where the logs are written (use `IIS AppPool\<app_pool_name>` to provide write permission).
+
+Debug level (`debugLevel`) values can include both the level and the location.
+
+Levels (in order from least to most verbose):
+
+* ERROR
+* WARNING
+* INFO
+* TRACE
+
+Locations (multiple locations are permitted):
+
+* CONSOLE
+* EVENTLOG
+* FILE
+
+The handler settings can also be provided via environment variables:
+
+* `ASPNETCORE_MODULE_DEBUG_FILE` &ndash; Path to the debug log file. (Default: *aspnetcore-debug.log*)
+* `ASPNETCORE_MODULE_DEBUG` &ndash; Debug level setting.
+
+> [!WARNING]
+> Do **not** leave debug logging enabled in the deployment for longer than required to troubleshoot an issue. The size of the log isn't limited. Leaving the debug log enabled can exhaust the available disk space and crash the server or app service.
+
+See [Configuration with web.config](#configuration-with-webconfig) for an example of the `aspNetCore` element in the *web.config* file.
+
+## Proxy configuration uses HTTP protocol and a pairing token
+
+*Only applies to out-of-process hosting.*
+
+The proxy created between the ASP.NET Core Module and Kestrel uses the HTTP protocol. Using HTTP is a performance optimization, where the traffic between the module and Kestrel takes place on a loopback address off of the network interface. There's no risk of eavesdropping the traffic between the module and Kestrel from a location off of the server.
```

Comment:
> Using HTTP is a performance optimization,

I don't think using HTTP is really a performance optimization. It's just what we happen to use. I would rephrase:

```suggestion
The proxy created between the ASP.NET Core Module and Kestrel uses the HTTP protocol where the traffic between the module and Kestrel takes place on a loopback address. There's no risk of eavesdropping the traffic between the module and Kestrel from a location off of the server.
```

Reply:
It's a design choice. We can exclude that line. The security risk line is the important part, people have asked about that.